src/wp-includes/html-api/class-wp-html-processor.php

```diff
@@ -623,6 +623,36 @@ private function step_in_body() {
 		$op_sigil = $this->is_tag_closer() ? '-' : '+';
 		$op       = "{$op_sigil}{$tag_name}";
 
+		if ( null === $tag_name && '#text' === $this->get_token_type() ) {
+			/*
+			 * This rule is necessary even without supporting text nodes in the
+			 * HTML Processor because the parser has to move past text nodes, and
+			 * there could be breadcrumb implications when the text triggers the
+			 * active format reconstruction.
+			 *
+			 */
+			$this->reconstruct_active_formatting_elements();
+
+			$current_token = $this->bookmarks[ $this->state->current_token->bookmark_name ];
+			if ( 1 === $current_token->length && "\x00" === $this->html[ $current_token->start ] ) {
```

Comment:
All null character tokens are supposed to be ignored.

- Should this be a `strlen(…) !== strspn( "\x00", …)` check like below for whitespace? 
- Should the tag processor not even stop on "text nodes" that are only null bytes? This is probably more correct.

Reply:
checked in 5b6e461bc544dbb4a8ad64f35e09631c7aed855a
I'm still leaving the quick test on the first character, hoping it will allow us to skip the `strspn` if possible, computing that value speculatively before it's accessed

Reply:
If we make that changes it's going to be much more involved and wrong sometimes (until we implement foreign content), isn't it? I agree, let's keep this as it.